processor/transformprocessor/internal/common/traces.go

```diff
@@ -0,0 +1,137 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/transformprocessor/internal/common"
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.uber.org/multierr"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/ottlresource"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/ottlscope"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/ottlspanevent"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/ottltraces"
+)
+
+type TracesContext interface {
+	ProcessTraces(ctx context.Context, td ptrace.Traces) error
+}
```

Comment:
Same question I asked, why not replace this with `consumer.Traces`

Reply:
@bogdandrutu I don't see the value of `type traceStatements []*ottl.Statement[ottltraces.TransformContext]` implementing a `Capabilities() Capabilities` function.  Also, I don't like how the API becomes `ConsumeTraces` instead of `ProcessTraces` as the function does not consume traces in the sense that the `consumer.Traces` API means.

Reply:
@bogdandrutu switched to `consumer` so you can see what it looks like.